fuse_optimizers/src/fixed_lag_smoother.cpp

```diff
@@ -268,6 +279,28 @@ void FixedLagSmoother::processQueue(fuse_core::Transaction& transaction)
   }
 }
 
+bool FixedLagSmoother::resetServiceCallback(std_srvs::Empty::Request& req, std_srvs::Empty::Response& res)
+{
+  started_ = false;
+  start_time_ = ros::TIME_MAX;
+  optimization_request_ = false;
+  // Clear all pending transactions
+  {
+    std::lock_guard<std::mutex> lock(pending_transactions_mutex_);
+    pending_transactions_.clear();
+  }
+  // Clear the graph and marginal tracking states
```

Comment:
And there's no weird state that we can get into from a transaction getting somehow inserted in between these two blocks? Just checking.

Reply:
I don't think so.
The ROS spinner is configured for a single thread. Since this is called from the ROS callback thread, no other ROS callbacks can fire while this is running. And thus new pending transactions cannot be added between blocks.